src/LegacyMongolidModel.php

```diff
@@ -0,0 +1,378 @@
+<?php
+namespace MongolidLaravel;
+
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Contracts\Validation\Factory as ValidationFactory;
+use Illuminate\Support\MessageBag;
+use Mockery;
+use Mockery\Expectation;
+use MongoDB\Collection;
+use Mongolid\Cursor\CursorInterface;
+use Mongolid\LegacyRecord;
+
+/**
+ * This class extends the Mongolid\LegacyRecord, so, in order
+ * to understand the ODM implementation make sure to check the
+ * base class.
+ *
+ * The MongolidLaravel\MongolidModel simply extends the original
+ * and framework agnostic model of MongoLid and implements some
+ * validation rules using Laravel validation components.
+ *
+ * Remember, this package is meant to be used with Laravel while
+ * the "leroy-merlin\mongolid" is meant to be used with other frameworks
+ * or even without any.
+ */
+abstract class LegacyMongolidModel extends LegacyRecord
+{
+    /**
+     * Validation rules.
+     *
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * Error message bag.
+     *
+     * @var MessageBag
+     */
+    protected $errors;
+
+    /**
+     * Public static mock.
+     *
+     * @var Mockery\Mock
+     */
+    public static $mock;
+
+    /**
+     * Public local mock.
+     *
+     * @var Mockery\Mock
+     */
+    public $localMock;
+
+    /**
+     * List of attribute names which should be hashed on save. For
+     * example: array('password');.
+     *
+     * @var array
+     */
+    protected $hashedAttributes = [];
+
+    /**
+     * Save the model to the database if it's valid. This method also
+     * checks for the presence of the localMock in order to call the save
+     * method into the existing Mock in order not to touch the database.
+     *
+     * @param bool $force force save even if the object is invalid
+     */
+    public function save(bool $force = false): bool
+    {
+        if ($this->localMockHasExpectationsFor('save')) {
+            return $this->getLocalMock()->save();
+        }
+
+        if ($force || $this->isValid()) {
+            $this->hashAttributes();
+
+            return parent::save();
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(): bool
+    {
+        $this->hashAttributes();
+
+        return parent::update();
+    }
+
+    /**
+     * Overwrites the delete method in order to be able to check for
+     * the expectation in the localMock in order to call the delete method
+     * into the existing mock and avoid touching the database.
+     */
+    public function delete(): bool
+    {
+        if ($this->localMockHasExpectationsFor('delete')) {
+            return $this->getLocalMock()->delete();
+        }
```

Comment:
```suggestion
        if ($this->localMockHasExpectationsFor('delete')) {
            $this->getLocalMock()->delete();
            
            return true;
        }
```
The same issue commented above.